C/common/storage_client.cpp

```diff
@@ -1498,17 +1612,87 @@ void StorageClient::handleException(const exception& ex, const char *operation,
 	// Now implement some recovery strategies
 	if (m_lastException.compare("Connection refused") == 0)
 	{
+		m_logger->error("Storage client: got Connection refused exception, retrying connecting to storage service");
+		int tries = 0;
+		int max_tries = 30;
+		
 		// This is probably because the storage service has gone down
 		if (m_management)
 		{
-			// Get a handle on the storage layer
+			PRINT_FUNC;
 			ServiceRecord storageRecord("Fledge Storage");
-			if (!m_management->getService(storageRecord))
+			do
+			{
+				PRINT_FUNC;
+				system("ps -ef | grep fledge.services.storage | grep -v grep | wc -l > /tmp/storage_service_instance_count");
```

Comment:
This is quite ugly, not very portable and makes the assumption the storage service is on the same machine as the client that is using it. We have been very careful not to add that restriction so that we may distribute services across machines or containers at some point.

I suspect you can find out what you need to by just using calls to the API and looking at error codes. A TCP connection to the API port can tell you if it is running or not.